src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -260,11 +260,12 @@ async def Generate(
                 kwargs["trace_headers"] = extract_trace_headers(headers)
             elif contains_trace_headers(headers):
                 log_tracing_disabled_warning()
+            unique_request_id = f"{request_id}-{i}"
             generators.append(
                 self.engine.generate(
                     inputs=inputs,
                     sampling_params=sampling_params,
-                    request_id=f"{request_id}-{i}",
+                    request_id=unique_request_id,
```

Comment:
Changs LGTM. One more NIT and then also squash the commits for the rebase merge.

NIT: We can remove the extra `unique_request_id` variable now after the other changes.